fili-core/src/test/groovy/com/yahoo/bard/webservice/util/IntervalUtilsSpec.groovy

```diff
@@ -326,6 +331,46 @@ class IntervalUtilsSpec extends Specification {
         DAY   | ["2012-02-03/2017-04-04"] | ["2012-02-03/2012-05-04", "2017-02-03/2017-04-04"]
     }
 
+    @Unroll
+    def "getCoarsestTimeGrain returns coarsest grain in the case of #description"() {
+        given:
+        PhysicalTableSchema schema1 = Mock(PhysicalTableSchema)
+        PhysicalTableSchema schema2 = Mock(PhysicalTableSchema)
+        PhysicalTableSchema schema3 = Mock(PhysicalTableSchema)
+
+        schema1.getTimeGrain() >> new ZonedTimeGrain(timeGrain1, DateTimeZone.forID(timeZone1))
+        schema2.getTimeGrain() >> new ZonedTimeGrain(timeGrain2, DateTimeZone.forID(timeZone2))
+        schema3.getTimeGrain() >> new ZonedTimeGrain(timeGrain3, DateTimeZone.forID(timeZone3))
+
+        PhysicalTable physicalTable1 = Mock(PhysicalTable)
+        PhysicalTable physicalTable2 = Mock(PhysicalTable)
+        PhysicalTable physicalTable3 = Mock(PhysicalTable)
+
+        physicalTable1.getSchema() >> schema1
+        physicalTable2.getSchema() >> schema2
+        physicalTable3.getSchema() >> schema3
+
+        expect:
+        IntervalUtils.getCoarsestTimeGrain(
+                [
+                        physicalTable1,
+                        physicalTable2,
+                        physicalTable3
+                ]
+        ).get() == new ZonedTimeGrain(expectedTimeGrain, DateTimeZone.forID(expectedTimeZone))
+
+        where:
+        timeGrain1 | timeGrain2 | timeGrain3 | timeZone1         | timeZone2             | timeZone3         | expectedTimeGrain | expectedTimeZone  | description
+        DAY        | DAY        | DAY        | 'America/Chicago' | 'America/Los_Angeles' | 'America/Phoenix' | DAY               | 'America/Chicago' | 'same grain but different time zones'
+        MINUTE     | HOUR       | DAY        | 'America/Phoenix' | 'America/Phoenix'     | 'America/Phoenix' | DAY               | 'America/Phoenix' | 'same time zone but different grans'
+        MINUTE     | HOUR       | DAY        | 'America/Chicago' | 'America/Los_Angeles' | 'America/Phoenix' | DAY               | 'America/Phoenix' | 'different grains and different time zones'
```

Comment:
Could we add some more cases that use week, month, quarter, and year? Some of those (ie. week) are odd in terms of how they line up, so that would be good to double-check.